src/pages/cart.tsx

```diff
@@ -0,0 +1,237 @@
+import React,{ FC, useEffect, useRef, useState } from "react";
+import Popup from "../components/Popup";
+import { getCartList, addCart } from "../scripts/addCart";
+import { useLocation } from "react-router";
+import { Link } from 'react-router-dom';
+import { setUrl, getParam } from "../scripts/setUrl";
+import Product from './product';
+interface product {
+    "id": number,
+    "title": string,
+    "description": string,
+    "price": number,
+    "discountPercentage":number,
+    "rating": number,
+    "stock": number,
+    "brand": string,
+    "category": string,
+    "thumbnail": string,
+    "images": string[]
+    "amount": number
+}
+
+const Cart: React.FunctionComponent<{clc: () => void;}> = (props) =>{
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const [test, settest] = useState<product[]>([]);
+    const [list, settList] = useState<product[]>([]);
+    const [maxElems, setMaxElems] = useState<number>(1);
+    const [pageIndex, setPageIndex] = useState<number>(1);
+    const [pagination, setPagination] = useState<number[]>([]);
+    const [products, setproductsLen] = useState<{length:number,total:number}>({length:0,total:0});
+    const [discount, setDiscount] = useState<number>(0);
+    const [promCodes, setPromCodes] = useState<{val: string; list: string[]}>({val: "",list:[]});
+    const [hidden, setHidden] = useState<boolean>(false);
+    const promo = useRef<HTMLDivElement>(null);
+    const promoInput = useRef<HTMLInputElement>(null);
+    const maxInput = useRef<HTMLInputElement>(null);
+
+    const {state} = useLocation();
+    
+    
+    useEffect(():void => {
+        const st = (state === null)?0:state.id
+        if(st > 0)show()
+        getList()
+        setMaxElems(2)
+        let products : product[] = JSON.parse(localStorage.getItem('cart')  + "")
+        settList(products)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [settest])
+
+    function show(){
+        setHidden(true)
+    }
+
+    useEffect(():void => {
+        let i: number = 0;
+        let p: number[] = [];
+        for(i = 0; i < Math.floor(list.length / maxElems); i++){
+            p.push(i+1);
+        }
+        if(Math.floor(list.length / maxElems) !== list.length / maxElems) p.push(i+1);
+        setPagination(p);
+
+        if(p.length  < pageIndex && pagination.length !== 0) {
+            setPageIndex(p.length)
+        }
+    }, [maxElems, list.length])
+
+    useEffect(():void => {
+        maxInput.current!.value = (getParam("max") === 'null')?'2':getParam("max")
```

Comment:
Are you sure that null should be a string here?

Reply:
yes